### PR TITLE
Proposed Amendment: We Maintain Code

### PR DESCRIPTION
**Pull request by JHWelch**
## Author checklist
- [x] I'm proposing a single change in this PR.
- I believe the proposed standard:
  - [x] benefits more than it costs us and
  - [x] doesn't conflict with our people-first value.
- [x] The standard's file(s) follow(s) [the established directory structure][directory-structure].
- I provided:
  - [x] a succinct summary,
  - [x] reasoning for the proposed change,
  - [x] an example/instructions on how to embody the standard unless repealing, and
  - [x] a longer explanation if appropriate.

*Check boxes if requirement met or irrelevant.*

## First reviewer checklist
***NOTE:** Some of the author's checklist items are intentionally omitted, because they are a matter of opinion and up for discussion with the whole team.*

- [x] This PR contains a single change.
- [x] The proposed standard follows [the established directory structure][directory-structure].
- The author provided:
  - [x] a succinct summary,
  - [x] reasoning for the proposed change, and
  - [x] an example/instructions on how to embody the standard unless repealing.

*Check boxes if requirement met or irrelevant.*

[directory-structure]: ../tree/master/standards#directory-structure

### CODE REVIEW

**Review by stuheiss — COMMENTED** [review]
Seems like the scope of this proposal is about framework version and language release version. I think of "Maintaining" as having a larger scope including libraries, refactoring, testing, etc. I wonder if framework and language release should be subsections of this larger concept of "Maintaining"

File: standards/practices/maintaining.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Maintaining
+
+## Summary
+We maintain our code and keep it updated whenever Highland is working on it.
```
Comment (curtisblackwell) [inline standards/practices/maintaining.md:4]: "keep it updated" feels a little vague. Do we want this to mean anything other than keep dependencies updated?

Should we say anything about latest major release or LTS or anything like that? Even if just to say it's a per-project decision that doesn't need to be explicit until it matters enough that it's causing pain?

**Review by curtisblackwell — COMMENTED** [review]
I think this looks great. I would consider a couple changes, but they're probably best considered as a group.

**Comment by JHWelch** [comment]
> Seems like the scope of this proposal is about framework version and language release version. I think of "Maintaining" as having a larger scope including libraries, refactoring, testing, etc. I wonder if framework and language release should be subsections of this larger concept of "Maintaining"

This sounds like a good point. I wrote the standard as "maintain" so we could generalize it later, but was focused on the update part. Maybe I should update this to "we update our dependencies".?  or keep it how it is with the understanding we will grow it.

File: standards/practices/maintaining.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Reasoning
+1. Over time it becomes harder and harder to work with code that is reliant on outdated language and framework versions.
+2. Keeping our software projects up to date mitigates the need to constantly be changing language and framework versions in development environments while moving between projects.
+3. The longer updates go unaddressed, the harder they become to make.
+4. Client's will almost never make code maintenance a priority, so we must. 
+
+## Example/Instructions
```
Comment (JHWelch) [inline standards/practices/maintaining.md:12]: Should probably add a section about dependabot in github.

**Comment by JHWelch** [comment]
Closing, but leaving branch for someone to look at in the future.

**Pull request closed without merging**